apps/stack/template.yaml

```diff
@@ -378,6 +398,22 @@ Resources:
           - UseStackName
           - !Sub "${AWS::StackName}-MetricStream"
           - !Sub "${NameOverride}-MetricStream"
+  MetricsPollerRole:
+    Type: AWS::Serverless::Application
+    Condition: EnableMetricsPollerRole
+    Properties:
+      Location: ../externalrole/template.yaml
+      NotificationARNs:
+        - !Ref Topic
+      Parameters:
+        ObserveAwsAccountId: !Ref ObserveAwsAccountId
+        AllowedActions: >-
+          cloudwatch:GetMetricData,cloudwatch:ListMetrics,tag:GetResources
+        DatastreamIds: !Join [",", !Ref DatastreamIds]
+        NameOverride: !If
+          - UseStackName
+          - !Sub "${AWS::StackName}-MetricsPollerRole"
```

Comment:
Beware that this dictates the name of the IAM role, which in turn must match the ARN the Observe poller was created with. Any mismatch in name will lead to traffic being ignored.

I believe that if you just pass `NameOverride` as is there will not be any clash with other IAM roles, but worth double checking.

Reply:
I want to avoid a clash between the metrics and resource poller, so I'm adding `-metrics-poller` as the suffix here and also in the place where the poller is created.